masu/processor/report_processor.py

```diff
@@ -0,0 +1,480 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Processor for Cost Usage Reports."""
+
+import csv
+import gzip
+import io
+import json
+import logging
+from itertools import islice
+from os import path
+
+from masu.config import Config
+from masu.database.report_db_accessor import ReportDBAccessor
+from masu.database.reporting_common_db_accessor import ReportingCommonDBAccessor
+from masu.processor.exceptions import MasuProcessingError
+
+LOG = logging.getLogger(__name__)
+
+ALLOWED_COMPRESSIONS = ('', 'gzip')
```

Comment:
I meant to bring this up. Because zip as a compression format can contain multiple files, I think it violates our process a single file at a time convention.  We can add it back in here and special case it, or figure out what to do with it elsewhere. Or move all compression handling outside of the processor.